src/services/services.ts

```diff
@@ -2110,6 +2111,14 @@ export function createLanguageService(
     function shouldGetType(sourceFile: SourceFile, node: Node, position: number): boolean {
         switch (node.kind) {
             case SyntaxKind.Identifier:
+                if (
+                    node.flags & NodeFlags.JSDoc && !isInJSFile(node) &&
+                    ((node.parent.kind === SyntaxKind.PropertySignature && (node.parent as PropertySignature).name === node) ||
+                        findAncestor(node, (n) => n.kind === SyntaxKind.Parameter))
+                    ) {
+                    // if we'd request type at those locations we'd get `errorType` that displays confusingly as `any`
+                    return false;
+                }

```

Comment:
I don't like how specific this is but I concluded that some condition like this is the best solution for https://github.com/microsoft/TypeScript/issues/57108 . I'm not 100% sure if I'm missing some other node types that should be ignored here or not though. The worst case scenario quick info will continue to display `any` when it shouldn't so I feel it's fine. If you come up with any extra situations that should be handled here I can tweak this and add more tests

Reply:
Note that it's still good to display `any` [here](https://github.com/microsoft/TypeScript/blob/353ccb7688351ae33ccf6e0acb913aa30621eaf4/tests/cases/fourslash/jsDocFunctionSignatures13.ts#L10) - and that's why I have to prevent the type request conditionally instead of just ignoring return values that are error types or smth like that